ref(profiling): fix tests as segment_id is not passed as part of the profile payload

viglia opened the pull request:
so the existing tests must work without segment_id being defined

#skip-changelog

Commit:
fix tests: segment_id is not passed as part of the profile payload

so the existing tests must work without segment_id being defined

Commit:
update changelog

Commit:
fix changelog

jjbayer reviewed (approved):
Thanks for following up! Could you please make sure that there is still at least one test that _does_ set `segment_id` on the input?

Commit:
add test were segment_id is set on the client side

we will ignore the segment_id set on the client side as we'll overwrite
with the right one, but the parsing should not fail even if it was set.

Commit:
Merge remote-tracking branch 'origin/master' into viglia/ref/test-optional-segment_id
